src/connectors/cloudtrail.py

```diff
@@ -0,0 +1,174 @@
+    , value:userIdentity.accountId::STRING user_identity_accountid
+    , value:userIdentity.invokedBy::STRING user_identity_invokedby
+    , value:userIdentity.accessKeyId::STRING user_identity_access_key_id
+    , value:userIdentity.userName::STRING user_identity_username
+    , value:userIdentity.sessionContext.attributes.mfaAuthenticated::STRING user_identity_session_context_attributes_mfa_authenticated
+    , value:userIdentity.sessionContext.attributes.creationDate::STRING user_identity_session_context_attributes_creation_date
+    , value:userIdentity.sessionContext.sessionIssuer.type::STRING user_identity_session_context_session_issuer_type
+    , value:userIdentity.sessionContext.sessionIssuer.principalId::STRING user_identity_session_context_session_issuer_principal_id
+    , value:userIdentity.sessionContext.sessionIssuer.arn::STRING user_identity_session_context_session_issuer_arn
+    , value:userIdentity.sessionContext.sessionIssuer.accountId::STRING user_identity_session_context_session_issuer_account_id
+    , value:userIdentity.sessionContext.sessionIssuer.userName::STRING user_identity_session_context_session_issuer_user_name
+    , value:errorCode::STRING error_code
+    , value:errorMessage::STRING error_message
+    , value:additionalEventData::VARIANT additional_event_data
+    , value:apiVersion::STRING api_version
+    , value:readOnly::BOOLEAN read_only
+    , value:resources::VARIANT resources
+    , value:serviceEventDetails::STRING service_event_details
+    , value:sharedEventID::STRING shared_event_id
+    , value:vpcEndpointID::STRING vpc_endpoint_id
+FROM DATA.{name}_STREAM, table(flatten(input => v:Records))
+WHERE ARRAY_SIZE(v:Records) > 0
+"""
+
+    comment = f"""
+---
+name: {name}
+"""
+
+    res = {}
+
+    db.create_stage(name=name + '_STAGE', url=bucket, prefix=prefix, cloud='aws',
+                    credentials=role, file_format=FILE_FORMAT)
+
+    db.create_table(name=name + '_STAGING', cols='(v variant)')
+
+    db.create_stream(name=name + '_STREAM', target=name+'_STAGING')
+
+    pipe_sql = f"COPY INTO DATA.{name}_STAGING FROM @DATA.{name}_STAGE/"
+    db.create_pipe(name=name + '_PIPE', sql=pipe_sql, replace=True)
+
+    db.create_table(name=name + "_EVENTS_CONNECTION", cols=CLOUDTRAIL_LANDING_TABLE, comment=comment)
+
+    db.create_task(name=name + '_TASK', schedule='15 minutes',
+                   warehouse=WAREHOUSE, sql=cloudtrail_ingest_task)
+
+    data = db.execute(f'DESC STAGE DATA.{name}_STAGE')
+    desc = list(data)
+    for i in desc:
+        if i[0] is 'STAGE_CREDENTIALS':
+            res[i[1]] = res[i[3]]
+
+    results = []
+    for field in res:
+        data = {'title': field, 'body': res[field]}
+        results.append(data)
+
+    results.append({'title': 'Next Steps', 'body': """
+Please follow the documentation at
+https://docs.snowflake.net/manuals/user-guide/data-load-s3-config.html#step-4-configuring-the-aws-iam-role-to-allow-access-to-the-stage
```

Comment:
we agreed to rewrite this in a way that makes more sense in the context of this flow?